### PR TITLE
Add test for #11383 (deadlock at initialization)

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build

File: root/core/recursive/fakelib.rootmap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[ fakelib.so ]
+# intentionally list the same lib twice
+class MyFakeClass
+
+[ anotherfakelib.so]
```
Comment (Axel-Naumann) [inline root/core/recursive/fakelib.rootmap:5]:
```suggestion
[ anotherfakelib.so ]
```

File: root/core/recursive/fakelib.rootmap [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[ fakelib.so ]
+# intentionally list the same lib twice
```
Comment (Axel-Naumann) [inline root/core/recursive/fakelib.rootmap:2]:
```suggestion
# list the same lib twice to trigger a warning during TCling()
```

**Review by Axel-Naumann — APPROVED** [review]
But see comments for possible improvements

**Pull request merged**